### PR TITLE
Feat/random goals and additionnal channel

### PR DESCRIPTION
**Pull request by MatthieuMoreau0**
Features added:
- Additionnal channel for current player controlled by agent
- Random goal at train time
- Evaluation of agent with set goal (to be able to compare between epochs on the same goal)

Refactored:
- Create env_step function for readability of train loop
- Add arguments for random goal
- rename create goal function

TODO:
- Add to logs the evaluation results
- Find pertinent value for nb of evaluation episodes
- Modify timesteps goal like in the paper (the three last timesteps are used) 
- Change length of each episode, in the paper it is 1 minute (525 steps, maybe use same length)

### CODE REVIEW

File: train.py [hunk 1]
```diff
@@ -18,7 +18,7 @@ def train(dfp_agent, env, optimizer, scheduler, args, list_opposition):
   # Buffer to compute rolling statistics 
   score_buffer = []
   r_t = 0
-  goal = utils.create_goal([10,0.2,0.1,2], len(args["TIMESTEPS"]))
+  goal = utils.create_last_timestep_goal([10,0.2,0.1,2], len(args["TIMESTEPS"]))
```
Comment (HugoSenetaire) [inline train.py]: Le but c'était pas que ça ce soit controllable en argument ?

File: train.py [hunk 2]
```diff
@@ -88,21 +89,23 @@ def train(dfp_agent, env, optimizer, scheduler, args, list_opposition):
...
         while not eval_episode_is_terminated:
-          action_op = agent(eval_observation[1]['observation'])
+          action_op = agent.get_action(observation[1]['observation'])
           eval_observation, action_dfp, sensory, measurements = env_step(eval_env, dfp_agent, eval_observation, action_op, eval_goal, args['CHANNEL_NAMES'], args['IMAGE_SIZE'], args['MEASUREMENT_NAMES'], epsilon=0, use_cuda=args['use_cuda'])
           eval_episode_is_terminated = eval_observation[0]['status'] == "DONE"
           episode_length+=1
         eval_rewards.append(eval_observation[0]['reward'])
         episode_lengths.append(episode_length)
         #TODO: save logs of evaluation
+      print(eval_rewards)
```
Comment (HugoSenetaire) [inline train.py]: Rajouter une description pour que ce soit lisible dans les prints (ou les logs)

**Review by HugoSenetaire — COMMENTED** [review]
Great Stuff

File: train.py (hunk 1 above)
Comment (MatthieuMoreau0) [reply train.py]: oui cf commits suivant

Comment (MatthieuMoreau0) [reply train.py]: vu ensemble, j'avais mal compris

**Pull request merged**